metamodel/processor/src/main/java/org/hibernate/search/metamodel/processor/annotation/processing/impl/ProcessorPropertyBindingProcessor.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.search.metamodel.processor.annotation.processing.impl;
+
+import java.util.Optional;
+
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.AnnotationValue;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.type.TypeMirror;
+import javax.tools.Diagnostic;
+
+import org.hibernate.search.engine.backend.document.DocumentElement;
+import org.hibernate.search.engine.backend.types.dsl.IndexFieldTypeFinalStep;
+import org.hibernate.search.mapper.pojo.bridge.PropertyBridge;
+import org.hibernate.search.mapper.pojo.bridge.binding.PropertyBindingContext;
+import org.hibernate.search.mapper.pojo.bridge.mapping.programmatic.PropertyBinder;
+import org.hibernate.search.mapper.pojo.bridge.runtime.PropertyBridgeWriteContext;
+import org.hibernate.search.mapper.pojo.mapping.definition.programmatic.PropertyMappingStep;
+
+public class ProcessorPropertyBindingProcessor extends AbstractProcessorAnnotationProcessor {
+	@Override
+	public void process(PropertyMappingStep mapping, AnnotationMirror annotation, Element element,
+			ProcessorAnnotationProcessorContext context) {
+		AnnotationMirror propertyBinder = getAnnotationProperty( annotation, "binder" );
+		if ( propertyBinder != null ) {
+			AnnotationValue type = getAnnotationValue( propertyBinder, "type" );
+			if ( type != null ) {
+				TypeMirror binder = (TypeMirror) type.getValue();
+
+				mapping.binder( new ProcessorPropertyBinder( context, binder ) );
+			}
+		}
```

Comment:
at first I thought we could just continue using the programmatic mapping to build the model from the "injected" fields, but that didn't work. So instead, I've added this binder ^

Reply:
Makes sense.